src/message/index.js

```diff
@@ -35,8 +35,9 @@ class BitswapMessage {
     }
   }
 
-  addBlock (block) {
-    this.blocks.set(mh.toB58String(block.key), block)
+  addBlock (block, hashAlg) {
+    const key = block.key(hashAlg)
```

Comment:
this is going to be async with the new crypto. so might be better to base it on top of that